src/System.Text.Json/src/Resources/Strings.resx

```diff
@@ -0,0 +1,198 @@
+            <xsd:complexType>
+              <xsd:attribute name="alias" type="xsd:string" />
+              <xsd:attribute name="name" type="xsd:string" />
+            </xsd:complexType>
+          </xsd:element>
+          <xsd:element name="data">
+            <xsd:complexType>
+              <xsd:sequence>
+                <xsd:element name="value" type="xsd:string" minOccurs="0" msdata:Ordinal="1" />
+                <xsd:element name="comment" type="xsd:string" minOccurs="0" msdata:Ordinal="2" />
+              </xsd:sequence>
+              <xsd:attribute name="name" type="xsd:string" use="required" msdata:Ordinal="1" />
+              <xsd:attribute name="type" type="xsd:string" msdata:Ordinal="3" />
+              <xsd:attribute name="mimetype" type="xsd:string" msdata:Ordinal="4" />
+              <xsd:attribute ref="xml:space" />
+            </xsd:complexType>
+          </xsd:element>
+          <xsd:element name="resheader">
+            <xsd:complexType>
+              <xsd:sequence>
+                <xsd:element name="value" type="xsd:string" minOccurs="0" msdata:Ordinal="1" />
+              </xsd:sequence>
+              <xsd:attribute name="name" type="xsd:string" use="required" />
+            </xsd:complexType>
+          </xsd:element>
+        </xsd:choice>
+      </xsd:complexType>
+    </xsd:element>
+  </xsd:schema>
+  <resheader name="resmimetype">
+    <value>text/microsoft-resx</value>
+  </resheader>
+  <resheader name="version">
+    <value>2.0</value>
+  </resheader>
+  <resheader name="reader">
+    <value>System.Resources.ResXResourceReader, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
+  </resheader>
+  <resheader name="writer">
+    <value>System.Resources.ResXResourceWriter, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
+  </resheader>
+  <data name="ArrayDepthTooLarge" xml:space="preserve">
+    <value>Depth of {0} within an array is larger than max depth of {1}.</value>
+  </data>
+  <data name="ArrayEndWithinObject" xml:space="preserve">
+    <value>We are within an object but observed an ']'.</value>
+  </data>
+  <data name="Default" xml:space="preserve">
+    <value>An error occurred while reading the json.</value>
+  </data>
+  <data name="DepthMustBePositive" xml:space="preserve">
+    <value>Mismatched number of start/end objects or arrays. Depth is {0} but must be greater than 0.</value>
+  </data>
+  <data name="EndOfCommentNotFound" xml:space="preserve">
+    <value>Expected either a '\n' or '*/' to indicate end of comment, but instead reached end of data.</value>
+  </data>
+  <data name="EndOfStringNotFound" xml:space="preserve">
+    <value>Expected a '"' to indicate end of string, but instead reached end of data.</value>
+  </data>
+  <data name="ExpectedDigitNotFound" xml:space="preserve">
+    <value>Invalid number. Last character read: '{0}'. Expected a digit ('0'-'9').</value>
```

Comment:
Presumably it expected a digit, a comma, or whitespace.  Maybe, depending on where you are in parsing, a decimal point.

I'd just drop the second sentence altogether.

Reply:
This exception is thrown after reading the sign byte (+/-), so I know explicitly that I expect a digit next and not some comma, white-space, etc. Similarly, if I have read valid digits, then a decimal point, I expect digits and not some other bytes that indicate end of token.

Reply:
If you think the detailed exception messages are worth keeping, I'd change the name of this one to `RequiredDigitNotFound`, because then it would lead a reviewer (at least me) to consider "where would it be _required_?"  My initial assumption was this was used generally during number parsing.

(The good part of resx files? They keep the text away from the code.  The bad part? They keep the text away from the code.)